MUI X v6 release announcement (#36398)
mui/material-ui: docs/pages/blog/mui-x-v6.md

---
title: MUI X v6 is ready for production!
description: Introducing the new major version of the advanced components.
date: 2023-03-06T00:00:00.000Z
authors: ['josefreitas']
card: true
tags: ['MUI X', 'News']
---

<img src="/static/blog/mui-x-v6/card.png" alt="" style="width: 100%; margin-bottom: 16px;" />

[MUI X v6.0.0](https://github.com/mui/mui-x/releases/tag/v6.0.0) is finally here with many improvements, new features, customization abilities, and a more robust foundation to accommodate the changes we want to deliver next.

## ✨ Features highlights ✨

## Data Grid

- [ApiRef now available for all users](/x/react-data-grid/api-object/).

> Manage pagination, scrolling, state, and other attributes through the Data Grid’s API object—previously available only in commercial plans, now accessible to all.

Author: Yeah, perhaps it makes more sense to use headers.

You can still use the Clock if you like to, but a replacement UI for the time picker is coming shortly after this release.

# Installation and migration

If this is your first try with MUI X, you can jump in the getting started section:

- [Data Grid](/x/react-data-grid/getting-started/)
- [Date Pickers](/x/react-date-pickers/getting-started/)

If you’re coming from previous versions, we recommend you check our migration guide with the complete list of breaking changes:

- [Data Grid](/x/migration/migration-data-grid-v5/)
- [Date Pickers](/x/migration/migration-pickers-v5/)

We also provide codemods to automate some of the necessary updates in your codebase. 

[code example using codemod]

# What’s next?

We operate in a continuous-delivery environment, so the plan is to keep rolling out new features in minor versions.

With the new base established in v6.0.0, we’ll continue our efforts to improve UI/UX, bring more customization abilities, and support new use cases.

Here’s a list of what you can expect to be delivered in the following months.

## Date Pickers

- Time Picker replacement for Clock
- Time Range
- Date Time Range

> - Month Range

Author: ```suggestion
- Ability to select a month range in the Date Range Picker
```

- Support for Joy UI

## Data Grid

- Clipboard importing
- Filtering on header
- Row Spanning
- Improved Panels
    - Column Management panel (column visibility, reordering, grouping, pinning, etc).
    - Filtering panel
- Support for Joy UI

## Charts 📊

- Preview of chart components 🔥

For more details, feel free to follow our [public roadmap](https://github.com/mui/mui-x/projects/1).

# Decoupling versions from MUI Core

> We’re officially decoupling MUI X’s versioning from MUI Core (including Material UI). We understand that this may cause confusion, so we’d like to explain the main reasons why we’re moving in this direction.

Reviewer: ```suggestion
We’re officially decoupling MUI X’s versioning from MUI Core (including `@mui/material`). We understand that this may cause confusion, so we’d like to explain the main reasons why we’re moving in this direction.
```

Maybe use the package name since "Material UI" is still used by a lot of people to refer to the entier "MUI" stack.

1. Soften migration pains with a yearly release cycle.
    
    The MUI X codebase is very dynamic, and the constant development of new features often requires more breaking changes. With a yearly release, we aim to deliver those in smaller, more digestible sizes, in a time window that you can rely on to make your plans and prepare for updates.
    

> 2. We aim to support not only Material UI but also Joy UI, and in the future, MUI Base as well.

Reviewer: ```suggestion
2. We aim to support not only Material UI but also Joy UI, and in the future, as well as MUI Base.
```

---

> title: MUI X v6 is ready for production!

Reviewer 2: I think that it would make more sense with. "ready for production!" makes sense from the context of the team and people using the beta, but it's likely a niche audience

```suggestion
title: Introducing MUI X v6.0.0
```

description: Introducing the new major version of the advanced components.
date: 2023-03-06T00:00:00.000Z
authors: ['josefreitas']
card: true
tags: ['MUI X', 'News']
---

<img src="/static/blog/mui-x-v6/card.png" alt="" style="width: 100%; margin-bottom: 16px;" />

[MUI X v6.0.0](https://github.com/mui/mui-x/releases/tag/v6.0.0) is finally here with many improvements, new features, customization abilities, and a more robust foundation to accommodate the changes we want to deliver next.

## ✨ Features highlights ✨

### Data Grid

> #### [ApiRef now available for all users](/x/react-data-grid/api-object/)

Reviewer 2: I think that it could be clearer than "all users" and we can hint that we moved it from the paid version, rather than we created a new API

```suggestion
#### [ApiRef was moved to the open-source version](/x/react-data-grid/api-object/)
```

> #### [ApiRef now available for all users](/x/react-data-grid/api-object/)

Reviewer 2: For the top of the blog, consider if it would make sense to copy the TOCs https://nextjs.org/blog/next-13. I like how it allows developers that have 3 minutes to get 80% of the information, while people that have 10 minutes to dive deeper.

Manage pagination, scrolling, state, and other attributes through the Data Grid's API object—previously available only in commercial plans, now accessible to all.

> The ApiRef enables developers to implement a whole new range of customizations that rely on programmatic control of the Grid's features.

Reviewer 2: We reference it like this in https://deploy-preview-36398--material-ui.netlify.app/x/react-data-grid/api-object/#how-to-use-the-api-object in the docs, should we do the same?

```suggestion
The `apiRef` enables developers to implement a whole new range of customizations that rely on programmatic control of the Grid's features.
```

```tsx
function CustomDataGrid(props) {
  const apiRef = useGridApiRef();

  return (
    <div>
      <Button onClick={() => apiRef.current.setPage(0)}>
        Go to page first page
      </Button>
      <DataGrid apiRef={apiRef} {...other} />
    </div>
  );
}
```

> 

Reviewer 2: Would it make sense to add a paragraph about why we are doing this move, making more open-source? We could also use the opportunity to link back to our stewardship https://mui-org.notion.site/Stewardship-542a2226043d4f4a96dfb429d16cf5bd.

These two features have been previously released in a minor v5 version under the `experimentalFeatures` flag.
Now they are officially stable.

### Date and Time Pickers

#### [Overhaul in the documentation](/x/react-date-pickers/)

The Date and Time Pickers documentation has drastically improved during pre-release, and it now features a more comprehensible navigation structure and many new examples.

#### [Improved layout customization](/x/react-date-pickers/custom-layout/)

Combining the slots concept with the grid layout, you can now rearrange, extend, and customize most of the sub-components used in the Pickers UI.

```tsx
function MyCustomLayout(props) {
  const { toolbar, tabs, content, actionBar } = usePickerLayout(props);

  // Put the action bar before the content
  return (
    <PickersLayoutRoot className={pickersLayout.root} ownerState={props}>
      {toolbar}
      {actionBar}
      <PickersLayoutContentWrapper className={pickersLayout.contentWrapper}>
        {tabs}
        {content}
      </PickersLayoutContentWrapper>
    </PickersLayoutRoot>
  );
}

> export default function CustomStaticDatePicker() {

Reviewer 2: ```suggestion

export default function CustomStaticDatePicker() {
```

Author: I removed this empty line to avoid scrolling in this demo box (which impacts the page scrolling).

Reviewer 2: This was added in https://github.com/mui/material-ui/pull/33100 because it's support hard to read when the code takes all the viewport. We tried to increase it a bit from 350 to 400 in https://github.com/mui/material-ui/pull/33100#discussion_r929066849. Maybe 450 would be even better?

  return <StaticDatePicker slots={{ layout: MyCustomLayout }} />;
}
```

> #### [Fields: the new default input for Pickers](/x/react-date-pickers/fields/)

Reviewer 2: I would personally vote to move it first under the date pickers section. It touches on a significant UX improvement for end-users, which is probably what developers ultimately care the most about.

> These rich text fields are specialized for date and time logic and offer quick navigation and isolated interaction within each section of a date value.

Reviewer 2: I think that there is an opportunity to describe the UX we had before, why it was crap, so people can better appreciate the value of this change.

Many users complained, and we completely agree, the clock view is not the ideal time-picking experience on Desktop, so we removed it as a default view.
You can still use the Clock if you like to, but a replacement UI for the time picker is coming shortly after this release.

## Installation and migration

If this is your first try with MUI X, you can jump in the getting started section:

- [Data Grid](/x/react-data-grid/getting-started/)
- [Date Pickers](/x/react-date-pickers/getting-started/)

If you're coming from previous versions, we recommend you check our migration guide with the complete list of breaking changes:

- [Data Grid](/x/migration/migration-data-grid-v5/)
- [Date Pickers](/x/migration/migration-pickers-v5/)

We also provide codemods to automate some of the necessary updates in your codebase.

```sh
npx @mui/x-codemod v6.0.0/preset-safe <path>
```

## What's next?

We operate in a continuous-delivery environment, so the plan is to keep rolling out new features in minor versions.

With the new base established in v6.0.0, we'll continue our efforts to improve UI/UX, bring more customization abilities, and support new use cases.

> Here's a list of what you can expect to be delivered in the following months.

Reviewer 2: Reading it, I felt that it could help to precise that we don't plan to stop at this list.

```suggestion
Here's a list of what you can expect to be delivered in the following months (what is next on our immediate roadmap).
```

### Data Grid

> - Clipboard importing

Reviewer 2: To consider if we should link the GitHub issue associated so developers can contribute more information on what they need.

- Filtering on header

> - Row Spanning

Reviewer 2: ```suggestion
- Row spanning
```

- Improved Panels
  - Column Management panel (column visibility, reordering, grouping, pinning, etc).
  - Filtering panel
- Support for Joy UI

### Date Pickers

> - Time Picker replacement for Clock

Reviewer 2: So it doesn't suggest that we will remove the clock on mobile?

```suggestion
- Time Picker replacement for the clock on desktop
```

- Time Range Picker
- Date Time Range Picker
- Ability to select a month range in the Date Range Picker
- Support for Joy UI

### Charts 📊

- Preview of chart components 🔥

You can get more details of our next steps in [MUI X public roadmap](https://github.com/mui/mui-x/projects/1).

## Decoupling versions from MUI Core

> We're officially decoupling MUI X's versioning from MUI Core (including `@mui/material`). We understand that this may cause confusion, so we'd like to explain the main reasons why we're moving in this direction.

Reviewer 2: ```suggestion
We're officially decoupling MUI X's versioning from MUI Core (including `@mui/material`: Material UI). We understand that this may cause confusion, so we'd like to explain the main reasons why we're moving in this direction.
```

> We're officially decoupling MUI X's versioning from MUI Core (including `@mui/material`: Material UI). We understand that this may cause confusion, so we'd like to explain the main reasons why we're moving in this direction.

Reviewer 2: Unbreakable space

```suggestion
We're officially decoupling MUI X's versioning from MUI Core (including `@mui/material`: Material UI). We understand that this may cause confusion, so we'd like to explain the main reasons why we're moving in this direction.
```

to fix

<img width="723" alt="Screenshot 2023-03-05 at 19 41 32" src="https://user-images.githubusercontent.com/3165635/222979437-9ceef3a2-ca27-4388-a3c0-8945442e5a3d.png">

---
title: MUI X v6 is ready for production!
description: Introducing the new major version of the advanced components.
date: 2023-03-06T00:00:00.000Z
authors: ['josefreitas']
card: true
tags: ['MUI X', 'News']
---

<img src="/static/blog/mui-x-v6/card.png" alt="" style="width: 100%; margin-bottom: 16px;" />

[MUI X v6.0.0](https://github.com/mui/mui-x/releases/tag/v6.0.0) is finally here with many improvements, new features, customization abilities, and a more robust foundation to accommodate the changes we want to deliver next.

## ✨ Features highlights ✨

## Data Grid

> - [ApiRef now available for all users](/x/react-data-grid/api-object/).

Reviewer 3: ```suggestion
- [ApiRef now available for all users](/x/react-data-grid/api-object/)
```

<img src="/static/blog/mui-x-v6/column-menu-custom-action.png" style="maxWidth: 692px; margin-bottom: 24px;" loading="lazy" alt="Column menu custom action" />

- [Cell selection](/x/react-data-grid/cell-selection/) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan)

Select a cell or group of cells like in an Excel sheet.
Cell selection is a powerful and flexible way to select data in the Data Grid.
It’s also the base for bulk editing and clipboard importing (coming soon).

<video style="margin-bottom: 5px; width: 100%;" autoplay muted loop playsinline >
  <source src="/static/blog/mui-x-v6/cell-selection.mp4" type="video/mp4" />
</video>

- Stable [Aggregation](/x/react-data-grid/aggregation/) [<span class="plan-premium"></span>](/x/introduction/licensing/#premium-plan) and [Row pinning](/x/react-data-grid/row-pinning/) [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)

These two features have been previously released in a minor v5 version under the experimentalFeatures flag.
Now they are officially stable.

## Date and Time Pickers

> - [Overhaul in the documentation.](/x/react-date-pickers/)

Reviewer 3: ```suggestion
- [Overhaul in the documentation](/x/react-date-pickers/)
```

The Date Pickers documentation has drastically improved during pre-release, and it now features a more comprehensible navigation structure and many new examples.

- [Improved layout customization](/x/react-date-pickers/custom-layout/)

Combining the slots concept with the grid layout, you can now rearrange, extend, and customize most of the sub-components used in the Pickers UI.

```tsx
function MyCustomLayout(props) {
  const { toolbar, tabs, content, actionBar } = usePickerLayout(props);

  // Put the action bar before the content
  return (
    <PickersLayoutRoot className={pickersLayoutClasses.root} ownerState={props}>
      {toolbar}
      {actionBar}
      <PickersLayoutContentWrapper
        className={pickersLayoutClasses.contentWrapper}
      >
        {tabs}
        {content}
      </PickersLayoutContentWrapper>
    </PickersLayoutRoot>
  );
}

export default function CustomStaticDatePicker() {
  return <StaticDatePicker slots={{ layout: MyCustomLayout }} />;
}
```

> - [Fields: the new default input for Pickers](/x/react-date-pickers/fields/).

Reviewer 3: ```suggestion
- [Fields: the new default input for Pickers](/x/react-date-pickers/fields/)
```
